### PR TITLE
Added FAQ section for entropy salt sources #298

### PR DESCRIPTION
**Pull request by YuraYelisieiev**
Fixes #	

 Changes:	
- Added FAQ section for entropy salt sources

 Does this change need to mentioned in CHANGELOG.md?	
yes | no	

 Requires testing	
no	

 Comments about testing, should you have some
Read and give me advice

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -22,7 +22,22 @@
     The salt is chosen in such a way that no two devices can generate the same value in (at least) the time frame of a human lifetime.
     Such values [might not be particularly random](https://security.stackexchange.com/questions/16117/in-hashing-does-it-matter-how-random-a-salt-is).
     Internal entropy is eventually combined with an external entropy received from the user host device.
-
+#### Salt sources
+- MCU core registers
+  * We use constant values of three registers, namely:
```
Comment (stdevAlDen) [inline FAQ.md]: @YuraYelisieiev are you trying to make this a list in th render? it's not working for me

I'm not sure that all the sources that you document are are implemented already, so it's ok in general, but needs some other review.

File: FAQ.md [hunk 2]
```diff
@@ -22,7 +22,22 @@
...
+    SP - track the call stack
+    LR - hold the address of the function.
+- Device UUID
+  * Every Microcontroller has its universal unique identifier which is stored at the specific location in memory, depending on different families.
+    As it is unique in every device, it can be used as a good salt for the entropy.
+- RTC(Real Time Clock) 
+- SysTick timer
+  * All the Microprocessor have it, regardless of the manufacturer. Systick timer is a part of the core. The dumbest and primitive timer.
+- TRNG
+  * is described above.
```
Comment (olemis) [inline FAQ.md]: how many random. bytes ? salted with entropy pool in advance or not ?

File: FAQ.md [hunk 3]
```diff
@@ -22,7 +22,22 @@
...
+#### Salt sources
+- MCU core registers
+  * We use constant values of three registers, namely: 
+    PC - shows which instruction perform next.
+    SP - track the call stack
+    LR - hold the address of the function.
+- Device UUID
+  * Every Microcontroller has its universal unique identifier which is stored at the specific location in memory, depending on different families.
+    As it is unique in every device, it can be used as a good salt for the entropy.
+- RTC(Real Time Clock)
```
Comment (olemis) [inline FAQ.md]: I'm not sure RTC is implemented , is it ?

File: FAQ.md [hunk 4]
```diff
@@ -22,7 +22,22 @@
     The salt is chosen in such a way that no two devices can generate the same value in (at least) the time frame of a human lifetime.
     Such values [might not be particularly random](https://security.stackexchange.com/questions/16117/in-hashing-does-it-matter-how-random-a-salt-is).
     Internal entropy is eventually combined with an external entropy received from the user host device.
-
+#### Salt sources
```
Comment (olemis) [inline FAQ.md]: Same relevance as `random source` please

File: FAQ.md [hunk 5]
```diff
@@ -22,7 +22,22 @@
...
+  * We use constant values of three registers, namely: 
+    PC - shows which instruction perform next.
+    SP - track the call stack
+    LR - hold the address of the function.
+- Device UUID
+  * Every Microcontroller has its universal unique identifier which is stored at the specific location in memory, depending on different families.
+    As it is unique in every device, it can be used as a good salt for the entropy.
+- RTC(Real Time Clock) 
+- SysTick timer
+  * All the Microprocessor have it, regardless of the manufacturer. Systick timer is a part of the core. The dumbest and primitive timer.
```
Comment (jdknives) [inline FAQ.md]: Not sure why we are using the word dumb here. @YuraYelisieiev

File: FAQ.md (hunk 3 above)
Comment (YuraYelisieiev) [reply FAQ.md]: It is not implemented, will remove it.

**Pull request merged**